WorkspaceServer/Kernel/INativeAssemblyLoadHelper.cs

```diff
@@ -1,11 +1,17 @@
 ﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System;
+using System.Collections.Generic;
+using System.IO;
+using WorkspaceServer.Packaging;
+
 namespace WorkspaceServer.Kernel
 {
-    public interface INativeAssemblyLoadHelper
+    public interface INativeAssemblyLoadHelper : IDisposable
     {
-        void Handle(string assembly);
-        void Configure(string v);
+        void Handle(ResolvedNugetPackageReference reference);
+
+        void SetNativeDllProbingPaths(IReadOnlyList<DirectoryInfo> nativeDllProbingPaths);
```

Comment:
(nit) `NativeDll` may not be the best name here since `.dll` is a Windows concept/extension.

How about `NativeLibrary`?